coderedcms/models/snippet_models.py

```diff
@@ -211,7 +211,36 @@ class Meta:
     def __str__(self):
         return self.name
 
+@register_snippet
+class ContentWall(models.Model):
+    """
+    Snippet that restricts access to a page with a modal.
+    """
```

Comment:
Should probably add a custom CSS class and ID that both get put in at the top-level div of the modal, so that the developer can easily target them with css/js if needed.